src/absil/il.fsi

```diff
@@ -727,16 +727,17 @@ type ILMethodBody =
       SourceMarker: ILSourceMarker option }
 
 /// Member Access
+// important: order of definition matters for AccessibilityLogic.GetILAccessOfILPropInfo
```

Comment:
We might want to avoid having to rely on the order of this.

Reply:
@TIHan , I hate it too, however, every time I tried to reproduce the actual broadening algorithm, it ended up being horrific.  One thing I think we could do, is to assign a specific value to each enum value, that way it is specific and deterministic, and not determined by the order in the source file.

Reply:
But relying on the order of declarations is how we F#. 😄

Reply:
@KevinRansom I think we can do this without having to do some broadening example.

Reply:
Lol you are not wrong ... but some developer who is not familiar with the ordering, might prefer to alphabetize the enum and break everything.
Arguably that is covered, by a comment that says, "the order is important".

Reply:
Can we cover it with a test instead of a comment (or maybe choose a testing approach that makes comments actually useful by verifying their correctness with regard to the code)?

Reply:
@jonsequitur I've just added such test in tests/fsharp/Compiler/ILMemberAccessTests.fs : https://github.com/dotnet/fsharp/pull/8354/files#diff-6415f4a32cd76160025147135e46baaa

If cases are added/removed, the test, if not updated, will fail, and if some of the existing comparisons don't hold true anymore, it will also fail.

Please let me know if that fits the bill.

Reply:
Roslyn also does ordering: http://sourceroslyn.io/#Microsoft.CodeAnalysis/Symbols/Accessibility.cs , http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/PropertyOrEventSymbolExtensions.cs,b31562460f5b74cc,references

I think for us, we technically don't need it for correctness in the compiler because the compiler simply asks a question if it is visible or not - all we need to do is ask if the get or set method is visible. However, the tooling, such as `Symbols.fs`, does need to know which is more accessible in terms of `ILMemberAccess`.

If we are going to do ordering, let's make the order just like Roslyn:
```fsharp
[<RequireQualifiedAccess>]
type ILMemberAccess =
    | CompilerControlled
    | Private
    | FamilyAndAssembly // protected and internal
    | Family // protected
    | Assembly // internal
    | FamilyOrAssembly // protected or internal
    | Public
```

Reply:
@TIHan I've updated it to match your order, thanks!